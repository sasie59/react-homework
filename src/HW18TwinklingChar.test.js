import React from "react";
import { render,act, cleanup } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import HW, {} from './HW18TwinklingChar';

afterEach(cleanup);

jest.useFakeTimers();

global.Math.random = () => 0;

test('should HW18TwinklingChar', () => {
  const {getByTestId} = render(<HW />);
  userEvent.type(getByTestId('input'),'8');
  expect(getByTestId('count').textContent).toBe('分數 : -1');
});

test('should HW18TwinklingChar', () => {
  const {getByTestId} = render(<HW />);
  userEvent.type(getByTestId('input'),'7');
  expect(getByTestId('count').textContent).toBe('分數 : -1');
});

test('should ', () => {
  act(() => {
    jest.useFakeTimers();
    const {container,getByTestId} = render(<HW />);
    let num ;
    jest.advanceTimersByTime(900);
    const isQ = getByTestId('isQ');
    num = isQ.textContent;
    expect(container).toMatchSnapshot();
  });
});
test('should ', () => {
  act(() => {
    jest.useFakeTimers();
    const {container,getByTestId} = render(<HW />);
    jest.advanceTimersByTime(1500);
    expect(getByTestId('isQ').textContent).toBe('-');
    jest.advanceTimersByTime(1500);
    expect(getByTestId('isQ').textContent).toBe('Q');
    expect(container).toMatchSnapshot();
  });
});

