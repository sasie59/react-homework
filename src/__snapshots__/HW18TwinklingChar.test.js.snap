// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  1`] = `
<div>
  <div>
    <h1
      data-testid="isQ"
    >
      -
    </h1>
    <h1
      data-testid="count"
    >
      分數 : 
      0
    </h1>
    <h2>
      出現Q按Q
    </h2>
    <input
      data-testid="input"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`should  2`] = `
<div>
  <div>
    <h1
      data-testid="isQ"
    >
      Q
    </h1>
    <h1
      data-testid="count"
    >
      分數 : 
      0
    </h1>
    <h2>
      出現Q按Q
    </h2>
    <input
      data-testid="input"
      type="text"
      value=""
    />
  </div>
</div>
`;
